src/main/java/com/zendesk/maxwell/MaxwellMysqlConfig.java

```diff
@@ -24,46 +32,112 @@ public MaxwellMysqlConfig() {
 		this.database = null;
 		this.user = null;
 		this.password = null;
+		this.sslMode = null;
 
-		this.jdbcOptions = new ArrayList<>();
-		this.jdbcOptions.add("zeroDateTimeBehavior=convertToNull");
-		this.jdbcOptions.add(String.format("connectTimeout=%d", connectTimeoutMS));
+		this.jdbcOptions = new HashMap<>();
+		this.jdbcOptions.put("zeroDateTimeBehavior", "convertToNull");
+		this.jdbcOptions.put("connectTimeout", String.valueOf(connectTimeoutMS));
 	}
 
-	public MaxwellMysqlConfig(String host, Integer port, String database, String user, String password) {
+	public MaxwellMysqlConfig(String host, Integer port, String database, String user, String password,
+			SSLMode sslMode) {
 		this.host = host;
 		this.port = port;
 		this.database = database;
 		this.user = user;
 		this.password = password;
+		this.sslMode = sslMode;
+	}
+
+	private void useSSL(boolean should) {
+		this.jdbcOptions.put("useSSL", String.valueOf(should));
+	}
+
+	private void requireSSL(boolean should) {
+		this.jdbcOptions.put("requireSSL", String.valueOf(should));
+	}
+
+	private void verifyServerCertificate(boolean should) {
+		this.jdbcOptions.put("verifyServerCertificate", String.valueOf(should));
 	}
 
 	public void setJDBCOptions(String opts) {
 		if (opts == null)
 			return;
 
 		for ( String opt : opts.split("&") ) {
-			this.jdbcOptions.add(opt.trim());
+			String[] valueKeySplit = opt.trim().split("=");
+			if (valueKeySplit.length == 2) {
+				this.jdbcOptions.put(valueKeySplit[0], valueKeySplit[1]);
+			}
+		}
+	}
+
+	private void setSSLOptions() {
+		if (sslMode != null && sslMode != SSLMode.DISABLED) {
+			this.useSSL(true); // for all SSL modes other than DISABLED, use SSL
+
+			this.verifyServerCertificate(false); // default to not verify server cert
+			this.requireSSL(false); // default to not require SSL
+
+			if (sslMode == SSLMode.REQUIRED || sslMode == SSLMode.VERIFY_CA
+					|| sslMode == SSLMode.VERIFY_IDENTITY) {
+				this.requireSSL(true);
+			}
+
+			if (sslMode == SSLMode.VERIFY_IDENTITY) {
+				this.verifyServerCertificate(true);
```

Comment:
same here, `this.verifyServerCertificate(sslMode == SSLMode.VERIFY_IDENTITY)`

minor, though, as written is fine too